nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSSession.java

```diff
@@ -899,8 +903,16 @@ protected void remove(String rootId) {
             }
         }
 
-        // remove all docs
-        transaction.removeStates(removedIds);
+        // remove root doc
+        transaction.removeStates(Collections.singleton(rootId));
+        Set<String> children = removedIds.stream().filter(id -> !id.equals(rootId)).collect(Collectors.toSet());
+        if (!children.isEmpty()) {
+            String nxql = String.format("SELECT * FROM Document WHERE ecm:uuid IN ('%s')",
+                    String.join("','", children));
```

Comment:
If there are even just 1000 or 10000 children doing a query with so many values in `IN` is not going to be possible due to backend limitations...

But the list of children ids was actually computed through `transaction.getDescendants` earlier so it's possible to have a query based on `ecm:ancestorId ` and the rootId. Note also that you should always do `SELECT * FROM Document, Relation` to really get all docs.